Intersect.Server/Entities/Npc.cs

```diff
@@ -32,6 +32,24 @@ public class Npc : Entity
         //Damage Map - Keep track of who is doing the most damage to this npc and focus accordingly
         public ConcurrentDictionary<Entity, long> DamageMap = new ConcurrentDictionary<Entity, long>();
 
+        // Returns the entity that ranks the highest on this NPC's damage map.
+        public Entity DamageMapHighest { 
+            get {
+                var damageMap = DamageMap;
+                long damage = 0;
+                Entity top = null;
+                foreach (var pair in damageMap)
```

Comment:
I think this can safely be made into a LINQ selector but I don't remember what the extension function is called.

Reply:
Would likely have to dive further into that, .Max does not work as it is not IComparable
.Max over the value only returns the value.
OrderByDescending().First() could work, but would likely be pissing slow.

Reply:
Looked into this a bit further, managed to do it with .Aggregate, but that is still about 50 times slower than the for loop on average.

OrderByDescending().FirstOrDefault() seems to range from between twice as slow to three times as slow.

The fancy schmancy solution people keep mention is MoreLinq's .MaxBy(), however that is slower even than the aggregate solution and takes an entire new library. So that isn't an option either.

The least horrible ''short'' version I've found so far seems to be OrderByDescending().FirstOrDefault().
Though keeping it as is is still faster.

Reply:
Making this an extension doesn't exactly speed it up.

It's about half a milisecond, and doesn't leverage OrderBy. Sure as heck aint pretty though.

`public static KeyValuePair<TKey, TValue> MaxByValue<TKey, TValue>(this IDictionary<TKey, TValue> dic) {
    var max = dic.Values.Max();
    var pair = dic.Where(d => d.Value.Equals(max)).FirstOrDefault();
    return new KeyValuePair<TKey, TValue>(pair.Key, pair.Value);
}`